tests/unit/cli/test_cli.py

```diff
@@ -184,3 +187,56 @@ def test_config_show_dict(runner, monkeypatch):
     assert "'DATA_DIR'" in result.output
     # using regex here, as output string may contain the color/formatting codes like "\x1b[32m"
     assert re.search(r"'DEBUG'[^:]*: [^']*True", result.output)
+
+
+@pytest.mark.parametrize(
+    "cli_input,expected_cmd",
+    [
+        ("status", "status"),
+        ("--debug status --format table", "status"),
+        ("config", "config"),
+        ("status with invalid args 💣", "status"),
+    ],
+)
+def test_publish_analytics_event_on_command_invocation(
+    cli_input, expected_cmd, runner, monkeypatch
+):
+    monkeypatch.setattr(localstack.cli.localstack, "ANALYTICS_API_RESPONSE_TIMEOUT_SECS", 3)
+    request_data = Queue()
+    input = cli_input.split(" ")
+
+    def handler(request, data):
+        request_data.put((request.__dict__, data))
+
+    with testutil.http_server(handler) as url:
+        monkeypatch.setenv("ANALYTICS_API", url)
+        monkeypatch.setattr(localstack.constants, "ANALYTICS_API", url)
+        runner.invoke(cli, input)
+        _, request_payload = request_data.get(timeout=5)
+
+    assert request_data.qsize() == 0
+    payload = json.loads(request_payload)
+    events = payload["events"]
+    assert len(events) == 1
+    event = events[0]
+    metadata = event["metadata"]
+    assert "client_time" in metadata
+    assert "session_id" in metadata
+    assert event["name"] == "cli_cmd"
+    assert event["payload"]["cmd"] == expected_cmd
+
+
+def test_disable_publish_analytics_event_on_command_invocation(runner, monkeypatch):
```

Comment:
awesome tests here! do you think we could also add one that asserts that the deadline/timeout is respected?